lib/usdUfe/ufe/Utils.cpp

```diff
@@ -400,7 +400,12 @@ bool allowedInStrongerLayer(
         ? stage->GetSessionLayer()
         : stage->GetRootLayer();
 
-    return getStrongerLayer(searchRoot, targetLayer, topLayer) == targetLayer;
```

Comment:
I'd like a comment explaining why we should allow when stronger is null. I know that historically, getting the logic and handling weird cases has been delicate, so I'd like to know when the stronger layer is null and why it is okay.

Reply:
Reading your test, I think it is because you target a layer within the reference?

We are presently considering allowing targeting layer in references, but we do not have a design yet for the UI nor how it would work, so if I'm correct that you're doing this because you want to target a layer in a reference, this is forward-looking code, but I think it may evolve and change behavior in the future.

In particular, I think we may enforce the same edit restrictions with the layer stack of the reference, instead of what would be a blanket "allow everything" that the null check implies. Let me know if my understanding is correct. It is not a blocker for your PR, but in the future it may affect your workflow in some cases if you were deleting prims with opinions in multiple layers in the reference and we no longer allow it in the future. (I suppose we could implement a special mode to allow it. Hard to know in advance.)

Reply:
Yes, it's mainly for targeting a layer in references. It's also the main intention of this PR. Alternatively, we could skip running `UsdUfe::applyCommandRestrictionNoThrow()` check altogether (which was the reason I modified `getStrongerLayer()`) if we receive an `UsdEditTarget` from the `EditRouter`. This way, we don't need to change the logic of `getStrongerLayer()` for now as it's meant for local layers. What do you think?

Reply:
Hey @pierrebai-adsk , any thoughts on this?

Reply:
I'd rather we keep calling it.

Reply:
Ok. I'll add comments to convey the intention of the changes.